adaptdl/adaptdl/torch/_metrics.py

```diff
@@ -105,24 +103,27 @@ def get_goodput_fn():
 
 def _fit_perf_params():
     state = _metrics_state()
-    items = state.profile.items()
-    items = [item for item in items if item[1]["count"] > 0]
-    keys = [item[0] for item in items]
-    values = [item[1] for item in items]
-    num_nodes, num_replicas, local_bsz, accumulation_steps = \
-        (np.array(val) for val in zip(*keys))
-    step_time = np.array([val["step_time"] / val["count"] for val in values])
-    sync_time = np.array([val["sync_time"] / val["count"] for val in values])
-    accumulation_time = np.array(
-            [val["accumulation_step_time"] / val["accumulation_count"]
-             if val["accumulation_count"] > 0 else 0.0
-             for val in values])
-    compute_time = step_time - sync_time
-    accumulation_time = np.where(
-        accumulation_steps > 0, accumulation_time, compute_time)
-    state.perf_params = fit_perf_params(
-        num_nodes, num_replicas, local_bsz, accumulation_steps,
-        step_time, compute_time, accumulation_time)
+    profile = {k: v for k, v in state.profile.items() if v.get("optim_count")}
+    # Convert profile into numpy arrays.
+    num_nodes, num_replicas, atomic_bsz = (np.array(k) for k in zip(*profile))
```

Comment:
I think the fact that `*profile` only takes the keys is a bit esoteric. Could you add that fact to the comment above?